src/models/callercalleemodel.cpp

```diff
@@ -39,6 +39,20 @@ CallerCalleeModel::CallerCalleeModel(QObject* parent)
         }
         emit dataChanged(index(0, Symbol), index(rowCount() - 1, Symbol));
     });
+
+    connect(Settings::instance(), &Settings::collapseTemplatesChanged, this, [this]() {
+        if (rowCount() == 0) {
```

Comment:
introduce a helper method and then call it here, pass the `this` model and then just take it as `QAbstractItemModel*` in the helper. Afaik you should be able to `emit dataChanged` from there too then